### PR TITLE
feat(REACTTODO-34): styling improve overall styling

### PR DESCRIPTION
**Pull request by macopul**

### CODE REVIEW

File: src/components/AddToDoSection/AddToDoSection.tsx [hunk 1]
```diff
@@ -11,16 +11,17 @@ const labels = {
...
-const AddToDoSection = ({ groupId, classname }: AddToDoSectionType) => {
+const AddToDoSection = ({ groupId, classname, buttonSectionClassname }: AddToDoSectionType) => {
   // now we can use useTaskStorage in any place at any level - cause methods inside are operating on shared context state which is one for all children - look inside TaskStoreContextProvider.tsx and in App.tsx
   const { addTask, addTaskGroup } = useTaskStorage();
   const [title, setTitle] = useState('');
 
   const handleAddButton = (action: string) => {
-    if (!title) {
-      window.alert('The task can not be empty');
+    if (title.trim() === '') {
```
Comment (woopul) [inline src/components/AddToDoSection/AddToDoSection.tsx]: Why not just  `if(!title.trim())` ? -  should be sufficient

File: src/components/TaskGroup/TaskGroup.module.scss [hunk 2]
```diff
@@ -15,36 +17,39 @@
...
 
 .scrolledList {
-  box-shadow: 0px 10px 10px -15px grey;
+  box-shadow: 0px 10px 10px -12px var(--primary-color-2);
 }
 
 .groupInput {
   font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva,
-  Verdana, sans-serif;
+    Verdana, sans-serif;
```
Comment (woopul) [inline src/components/TaskGroup/TaskGroup.module.scss]: unnecessary space ?

File: src/components/TaskGroup/TaskGroup.tsx [hunk 3]
```diff
@@ -33,6 +34,10 @@ const TaskGroup = ({ taskList, groupTitle, groupId, isHidden }: TaskItemGroupTyp
   const taskQuantity = groups.find((group) => group.groupId === groupId)!.taskList.length;
   const [taskCount, setTaskCount] = useState(taskQuantity);
 
+  useEffect(() => {
+    groupRef.current?.scrollIntoView({ behavior: 'smooth' });
```
Comment (woopul) [inline src/components/TaskGroup/TaskGroup.tsx:38]: Is that added for each task group in the list and scrolls one by one to the view during render ?

File: src/components/TaskItem/TaskItem.tsx [hunk 4]
```diff
@@ -13,10 +13,15 @@ const TaskItem = ({ title, checked, id, groupId }: TaskItemType) => {
...
   const [isTaskChecked, setIsTaskChecked] = useState(checked);
   const [isTaskEditable, setIsTaskEditbale] = useState(false);
+  const taskRef = useRef<HTMLDivElement>(null);
   const ref = useRef<HTMLInputElement>(null);
   useClickOutside(ref, () => setIsTaskEditbale(false), `ignoreClickOutside${id}`);
   const { updateTask, deleteTask } = useTaskStorage();
 
+  useEffect(() => {
+    taskRef.current?.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+  }, []);
```
Comment (woopul) [inline src/components/TaskItem/TaskItem.tsx:23]: Same as the above but on the task item level - What would you like to achieve here by this useEffect?
Cause remember that fires on mount for every individual task

Comment (macopul) [reply src/components/TaskItem/TaskItem.tsx:23]: Yes, on mount - just this first time, to scroll if view is in other area.

File: src/components/AddToDoSection/AddToDoSection.tsx (hunk 1 above)
Comment (macopul) [reply src/components/AddToDoSection/AddToDoSection.tsx]: yes, that is right. Thanks for comment

File: src/components/TaskGroup/TaskGroup.module.scss (hunk 2 above)
Comment (macopul) [reply src/components/TaskGroup/TaskGroup.module.scss]: removed

**Review by woopul — APPROVED** [review]

**Pull request merged**